src/OpenRiaServices.EntityFramework/Framework/DbDomainService.cs

```diff
@@ -304,5 +304,68 @@ private static async Task SetChangeSetConflictsAsync(Dictionary<DbEntityEntry, C
                 }
             }
         }
+
+        /// <summary>
+        /// Mark the entity as added
+        /// </summary>
+        /// <typeparam name="TEntity">Entity type</typeparam>
+        /// <param name="entity">The modified entity</param>
+        /// <param name="dbSet">The db set</param>
+        protected void AttachAsAdded<TEntity>(TEntity entity, DbSet<TEntity> dbSet = null)
+            where TEntity : class
+        {
+            EntityEntry<TEntity> entityEntry = DbContext.Entry(entity);
+
+            if (entityEntry.State != EntityState.Detached)
+            {
+                entityEntry.State = EntityState.Added;
+            }
+            else
+            {
+                dbSet ??= DbContext.Set<TEntity>();
+                dbSet.Add(entity);
+            }
+        }
+
+        /// <summary>
+        /// Mark the object for removal
+        /// </summary>
+        /// <typeparam name="TEntity">Entity type</typeparam>
+        /// <param name="entity">The entity to remove</param>
+        /// <param name="dbSet">The db set</param>
+        protected void AttachAsDeleted<TEntity>(TEntity entity, DbSet<TEntity> dbSet = null)
+            where TEntity : class
+        {
+            EntityEntry<TEntity> entityEntry = DbContext.Entry(entity);
+
```

Comment:
Todo: review again. 

1. Should if statement check detached status?

2. In the else statement we could maybe [attach](https://learn.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.dbcontext.attach?view=efcore-8.0#microsoft-entityframeworkcore-dbcontext-attach-1(-0))) and remove directly using DbContext (and maybe entiryEntry) directly.
3. If so maybe we should remove dbset instead of making it optional as in my original suggestion)